lms/djangoapps/program_enrollments/rest_api/v1/utils.py

```diff
@@ -111,10 +168,143 @@ def wrapped_function(self, request, **kwargs):
                 developer_message="the program's curriculum does not contain the given course",
                 error_code='course_not_in_program'
             )
-        return view_func(self, request, **kwargs)
+        return view_func(self, *args, **kwargs)
     return wrapped_function
 
 
+def verify_user_enrolled_in_program(view_func):
+    """
+    Raised PermissionDenied if the `target_user` is not enrolled in the program.
+
+    Expects to be used within a UserProgramViewMixin subclass.
+    """
+    @wraps(view_func)
+    def wrapped_function(self, *args, **kwargs):
+        """
+        Wraps the given view_function.
+        """
+        user_enrollment_qs = fetch_program_enrollments(
+            program_uuid=self.program_uuid,
+            users={self.target_user},
+            program_enrollment_statuses={ProgramEnrollmentStatuses.ENROLLED},
+        )
+        if not user_enrollment_qs.exists():
+            raise PermissionDenied
+        return view_func(self, *args, **kwargs)
+    return wrapped_function
+
+
+def get_enrollments_for_courses_in_program(user, program):
+    """
+    Get a user's active enrollments for course runs with the given program.
+
+    Note that this is distinct from the user's *program course enrollments*,
+    which refers to courses that were enrollmed in *through* a program.
+
+    In the case of this function, the course runs themselves must be part of the
+    program, but the enrollments do not need to be associated with a program enrollment.
+
+    Arguments:
+        user (User)
+        program (dict)
+
+    Returns QuerySet[CourseEnrollment]
+    """
+    course_keys = [
+        CourseKey.from_string(key)
+        for key in course_run_keys_for_program(program)
+    ]
+    return CourseEnrollment.objects.filter(
+        user=user,
+        course_id__in=course_keys,
+        mode__in=[CourseMode.VERIFIED, CourseMode.MASTERS],
+        is_active=True,
+    )
+
+
+def get_enrollment_overviews(user, program, enrollments, request):
```

Comment:
The name should be `get_enrollments_overview`?

Reply:
I am thinking of an "enrollment overview" as a dictionary describing an enrollment, so the function is getting a list of enrollment overviews. I based that thinking on the existing serializer ProgramCourseEnrollmentOverview.

Reply:
Thanks for the explanation.